pyccel/parser/semantic.py

```diff
@@ -1381,14 +1382,18 @@ def _assign_lhs_variable(self, lhs, d_var, rhs, **settings):
 
                 # Not yet supported for arrays: x=y+z, x=b[:]
                 # Because we cannot infer shape of right-hand side yet
-                know_lhs_shape = lhs.shape or (lhs.rank == 0) \
+                know_lhs_shape = all(sh is not None for sh in lhs.alloc_shape) \
+                        or (lhs.rank == 0) \
                         or isinstance(rhs, (Variable, EmptyLike, DottedVariable))
                 if not know_lhs_shape:
                     msg = "Cannot infer shape of right-hand side for expression {} = {}".format(lhs, rhs)
                     errors.report(PYCCEL_RESTRICTION_TODO+'\n'+msg,
                         bounding_box=(self._current_fst_node.lineno, self._current_fst_node.col_offset),
                         severity='fatal', blocker=self.blocking)
-
+                elif lhs.rank>0 and isinstance(rhs, PyccelOperator):
+                    #TODO: Provide order once issue #335 is fixed
+                    #stmts.append(Assign(lhs, Empty(lhs.alloc_shape, dtype, lhs.order)))
```

Comment:
I know this is just a comment, but nevertheless... shouldn't `stmts` be changed to `new_expressions`?

Reply:
You're right. I had stmts as a local variable at one point. I must have missed this when I changed it to new_expressions